### PR TITLE
[DEPRECATION] #99098 - Static usage of FormProtectionFactory

### PR DESCRIPTION
**Pull request by brotkrueml**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/266
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 1]
```diff
@@ -17,69 +25,44 @@ request forgery (CSRF).
...
-
-    // use TYPO3\CMS\Core\FormProtection\FormProtectionFactory;
-
-    FormProtectionFactory::get()
-        ->persistTokens();
-
-In backend lists, it might be necessary to generate hundreds of tokens.
-So, the tokens are not automatically persisted after creation for performance
-reasons.
+    $formToken = $formProtection->generateToken('tt_content', 'edit', (string)$uid);
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:53]: So `$formProtection` is a FormProtectionFactory that got injected? Then it should be called with $this and we should show how to inject it including the class it has somewhere. Otherwise this codesnippets gets hard to understand

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 2]
```diff
@@ -17,69 +25,44 @@ request forgery (CSRF).
...
 
-When processing the data that has been submitted by the form,
-you can check that the form token is valid like this:
+When processing the data that has been submitted by the form, you can check that
+the form token is valid like this:
 
 ..  code-block:: php
 
     // use TYPO3\CMS\Core\FormProtection\FormProtectionFactory;
     // use TYPO3\CMS\Core\Utility\GeneralUtility;
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:61]: We dont need this import now anymore

File: Documentation/ApiOverview/FormProtection/Index.rst [hunk 3]
```diff
@@ -89,54 +72,14 @@ you can check that the form token is valid like this:
         // create a flash message for an invalid token
     }
 
-
-..  note::
-    The :php:`validateToken()` method invalidates the token with the token ID.
```
Comment (linawolf) [inline Documentation/ApiOverview/FormProtection/Index.rst:94]: Is this not true anymore? Why did you remove the note?

File: Documentation/ApiOverview/FormProtection/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/FormProtection/Index.rst:53]: Up there is, see the literalinclude. FormProtectionFactory is injected, and formProtection is retrieved from that factory in the action (or whatever is used). These snippets below are just derived from that class above. Using a full class will blow up the examples.

File: Documentation/ApiOverview/FormProtection/Index.rst (hunk 3 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/FormProtection/Index.rst:94]: Nope, the method does not exist anymore. Also - when looking at the code - the token is just validated from the input and the beuser session.

I will provide a follow-up for 11.5 for this part.

I tried to build an implementation for examples ext, but can't figure out an easy scenario. When creating a link with a be view helper (like for a form action) the token is already there, so no need to create it and validate on your own.

File: Documentation/ApiOverview/FormProtection/Index.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/FormProtection/Index.rst:53]: Let's maybe put in the description that this is an excerpt of FormProtectionExample.php?

**Review by linawolf — APPROVED** [review]
